File: src/STech.Client/ClientApp/src/app/shared/models/product.ts
import {ProductRating} from "./productRatings";
import {ICategory} from "./category";
import {IPicture} from "./picture";

export interface IProduct {
  productID: number,
  id: number,
  name: string,
  description: string,
  price: number,
  stockQuantity: number,
  pictureUrls: string[],
  categoryName: string,
  ratings: ProductRating
}

export interface IProductList {
  products: IProduct[]
}

export interface IProductDetails {
  id: number;
  name: string;
  description: string;
  price: number;
  cost: number;
  stockQuantity: number;
  categoryID: number;
  category: ICategory;
  productPictures: IProductPictures[];
}

export interface IProductPictures {
  id: number;
  productID: number;
  pictureID: number;
  picture: IPicture;
}

File: src/STech.Core/Domain/Entities/Product.cs
namespace STech.Core.Domain.Entities;

public class Product : BaseEntity
{
    public decimal Price { get; set; }
    public decimal Cost { get; set; }
    public int StockQuantity { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public int CategoryID { get; set; }
    public virtual Category Category { get; set; }
    public virtual List<ProductPicture> ProductPictures { get; set; }
}